### PR TITLE
Feat/lib 1271 fzlayout

### PR DESCRIPTION
**Pull request by Trebh**
Jira: [LIB-1271](https://fiscozen.atlassian.net/browse/LIB-1271)

[LIB-1271]: https://fiscozen.atlassian.net/browse/LIB-1271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/layout/src/__tests__/FzLayout.spec.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+describe.concurrent('FzLayout', () => {
+  it('matches snaphost', async ({ expect }) => {
+    const wrapper = mount(FzLayout, {
+      props: {},
+      slots: {}
+    })
+
+    expect(wrapper.html()).toMatchSnapshot()
+  })
```
Comment (Crish15) [inline packages/layout/src/__tests__/FzLayout.spec.ts:13]: I would add other tests based on the values the layout can take

File: packages/layout/src/FzLayout.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            </div>
+            <div class="col-span-2">
+                <slot name="header"></slot>
+            </div>
+            <div class="col-span-2 lg:col-span-1">
+                <slot></slot>
+            </div>
+            <div class="col-span-2 lg:col-span-1">
+                <slot name="right-shoulder"></slot>
+            </div>
```
Comment (oxcened) [inline packages/layout/src/FzLayout.vue:32]: I would rather have a single template for each layout, it's more redundant but makes it more clear and easier to maintain
There's also two default slots for `rightShoulderNavbar`

File: packages/layout/src/FzLayout.vue [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            res = 'grid-rows-[40px_160px_60px_1fr] grid-cols-1 lg:grid-rows-[40px_160px_1fr] lg:grid-cols-[260px_1fr]'
+            break;
+        case 'rightShoulderNavbar':
+            res = 'grid-rows-[40px_160px_60px_1fr] grid-cols-1 lg:grid-rows-[40px_160px_1fr] lg:grid-cols-[1fr_260px]'
+            break;
+        case 'squares':
+            res = 'grid-cols-1 grid-rows-auto sm:grid-cols-2 sm:grid-rows-auto md:grid-cols-3 md:grid-rows-auto auto-rows-[332px]'
+            break;
+        default:
+            break;
```
Comment (oxcened) [inline packages/layout/src/FzLayout.vue:67]: I'd just return those strings. no need for res

File: packages/layout/src/FzLayout.vue (hunk 2 above)
Comment (Trebh) [reply packages/layout/src/FzLayout.vue:32]: I'll fix the double slot. As per the single template, I actually prefer this way at least for "default slot - only" templates.

File: packages/layout/src/FzLayout.vue (hunk 3 above)
Comment (Trebh) [reply packages/layout/src/FzLayout.vue:67]: I very much prefer functions to have a single exit if possible, except for quick "out" conditions at the very top when needed.

**Review by oxcened — APPROVED** [review]

**Review by Crish15 — APPROVED** [review]

**Pull request merged**